File: cpp/leetcode/293.cpp
class Solution {
public:
    vector<string> generatePossibleNextMoves(string s) {
        vector<string> moves;
        int n = s.length();
        for (int i = 0; i < n - 1; i++) {
            if (s[i] == '+' && s[i + 1] == '+') { 
                moves.push_back(s.substr(0, i) + "--" + s.substr);
            }
        }
        return moves;
    }
};

File: python/leetcode/293.py
class Solution(object):
    def generatePossibleNextMoves(self, s):
        n, res = len(s), []
        for i in range(n - 1):
            if s[i:i+2] == '++':
                res.append(s[:i] + '--' + s[i+2:])
        return res